File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/java/dmn/DecisionService_002.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"ds.ftl", "decisionService_002"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "decisionService_002",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class DecisionService_002 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "decisionService_002",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
        com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private static class DecisionService_002LazyHolder {
        static final DecisionService_002 INSTANCE = new DecisionService_002();
    }
    public static DecisionService_002 instance() {
        return DecisionService_002LazyHolder.INSTANCE;
    }

    private final Decision_002 decision_002;

    private DecisionService_002() {
        this(new Decision_002());
    }

    private DecisionService_002(Decision_002 decision_002) {
        this.decision_002 = decision_002;
    }

    @java.lang.Override()
    public String applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(input_.get("decision_002_input"), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'DecisionService_002'", e);
            return null;
        }
    }

    public String apply(String decision_002_input, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start DS 'decisionService_002'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long decisionService_002StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments decisionService_002Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            decisionService_002Arguments_.put("decision_002_input", decision_002_input);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, decisionService_002Arguments_);

            // Bind input decisions
            cache_.bind("decision_002_input", decision_002_input);

            // Evaluate DS 'decisionService_002'
            String output_ = lambda.apply(decision_002_input, context_);

            // End DS 'decisionService_002'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, decisionService_002Arguments_, output_, (System.currentTimeMillis() - decisionService_002StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'decisionService_002' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<String> lambda =
        new com.gs.dmn.runtime.LambdaExpression<String>() {
            public String apply(Object... args_) {
                String decision_002_input = 0 < args_.length ? (String) args_[0] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 1 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[1] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                String decision_002 = DecisionService_002.this.decision_002.apply(context_);

                return decision_002;
            }
    };
}

File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/python/dmn/DecisionService_002.py
import typing
import decimal
import datetime
import isodate
import time

import jdmn.runtime.Context
import jdmn.runtime.DefaultDMNBaseDecision
import jdmn.runtime.ExecutionContext
import jdmn.runtime.LambdaExpression
import jdmn.runtime.LazyEval
import jdmn.runtime.Pair
import jdmn.runtime.Range
import jdmn.runtime.RuleOutput
import jdmn.runtime.RuleOutputList

import jdmn.runtime.annotation.Annotation
import jdmn.runtime.annotation.AnnotationSet
import jdmn.runtime.annotation.DRGElementKind
import jdmn.runtime.annotation.ExpressionKind
import jdmn.runtime.annotation.HitPolicy

import jdmn.runtime.cache.Cache

import jdmn.runtime.external.ExternalFunctionExecutor

import jdmn.runtime.listener.Arguments
import jdmn.runtime.listener.DRGElement
import jdmn.runtime.listener.EventListener
import jdmn.runtime.listener.Rule

import Decision_002


# Generated(value = ["ds.ftl", "decisionService_002"])
class DecisionService_002(jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision):
    DRG_ELEMENT_METADATA: jdmn.runtime.listener.DRGElement.DRGElement = jdmn.runtime.listener.DRGElement.DRGElement(
        "http://www.montera.com.au/spec/DMN/0085-decision-services",
        "decisionService_002",
        "",
        jdmn.runtime.annotation.DRGElementKind.DRGElementKind.DECISION_SERVICE,
        jdmn.runtime.annotation.ExpressionKind.ExpressionKind.OTHER,
        jdmn.runtime.annotation.HitPolicy.HitPolicy.UNKNOWN,
        -1
    )
    _instance = None

    def __init__(self):
        raise RuntimeError("Call instance() instead")

    @classmethod
    def instance(cls):
        if cls._instance is None:
            cls._instance = cls.__new__(cls)
            jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision.__init__(cls._instance)
            cls._instance.initSubDecisions()
        return cls._instance

    def initSubDecisions(self, decision_002: Decision_002.Decision_002 = None):
        self.decision_002 = Decision_002.Decision_002() if decision_002 is None else decision_002

    def apply(self, decision_002_input: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[str]:
        try:
            # Start DS 'decisionService_002'
            annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet = None if context_ is None else context_.annotations
            eventListener_: jdmn.runtime.listener.EventListener.EventListener = None if context_ is None else context_.eventListener
            externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor = None if context_ is None else context_.externalFunctionExecutor
            cache_: jdmn.runtime.cache.Cache.Cache = None if context_ is None else context_.cache
            decisionService_002StartTime_ = int(time.time_ns()/1000)
            decisionService_002Arguments_ = jdmn.runtime.listener.Arguments.Arguments()
            decisionService_002Arguments_.put("decision_002_input", decision_002_input)
            eventListener_.startDRGElement(self.DRG_ELEMENT_METADATA, decisionService_002Arguments_)

            # Bind input decisions
            cache_.bind("decision_002_input", decision_002_input)

            # Evaluate DS 'decisionService_002'
            output_: typing.Optional[str] = self.evaluate(decision_002_input, context_)

            # End DS 'decisionService_002'
            eventListener_.endDRGElement(self.DRG_ELEMENT_METADATA, decisionService_002Arguments_, output_, (int(time.time_ns()/1000) - decisionService_002StartTime_))

            return output_
        except Exception as e:
            self.logError("Exception caught in 'decisionService_002' evaluation", e)
            return None

    def evaluate(self, decision_002_input: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[str]:
        # Apply child decisions
        decision_002: typing.Optional[str] = self.decision_002.apply(context_)

        return decision_002
